src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -121,6 +123,37 @@ public static function noConstraints(Closure $callback)
         }
     }
 
+    /**
+     * Conditionally modify the relation.
+     *
+     * Unlike the query builder's when method, we always return the relation
+     * instance so that relationship-specific methods remain available.
+     */
+    public function when($value, callable $callback, callable $default = null)
+    {
+        if ($this instanceof BelongsToMany) {
+            // For BelongsToMany, run the callback on the relation instance.
+            if ($value) {
+                $callback($this, $value);
+            } elseif ($default) {
+                $default($this, $value);
+            }
+            return $this;
+        }
+
+        if ($this instanceof MorphTo) {
+            // For MorphTo, update the underlying query builder and set it back on the relation.
+            $query = $this->getQuery()->when($value, $callback, $default);
```

Comment:
Isn't the code inside the `if ($this instanceof MorphTo)` the same as the code right below it? So can't this whole if statement be removed?